pdc/apps/auth/views.py

```diff
@@ -73,8 +75,58 @@ def logout(request):
     return HttpResponseRedirect(redirect_to)
 
 
+def get_resource_permission_set(user):
+    resource_permission_set = set([])
+    if user.is_superuser:
+        resource_permission_set = set([obj for obj in models.ResourcePermission.objects.all()])
+    else:
+        group_id_list = [group.id for group in user.groups.all()]
+        queryset = models.GroupResourcePermission.objects.filter(group__id__in=group_id_list)
+
+        for group_resource_permission in queryset:
+            resource_permission_set.add(group_resource_permission.resource_permission)
+    return resource_permission_set
+
+
+def _change_row(row, value, d=[]):
+    http_method = ['create', 'read', 'update', 'delete']
+    for v in d:
+        try:
+            index = http_method.index(v.lower())
+            row[index + 1] = value
+        except ValueError:
+            raise Exception("The action %s is not valid" % v)
+    return row
+
+
+def _get_resource_permissions_matrix(user):
+    d = {}
```

Comment:
Could name `d` a more meaningful name? It's like a dict map resource name to the resource's permission list. A meaningful name looks make code easier to understand.